core/model/src/test/java/org/eclipse/rdf4j/model/util/ValuesTest.java

```diff
@@ -508,6 +509,18 @@ public void testTemporalAccessorLiteralNull() {
 
 	}
 
+	@Test
+	public void testTemporalAccessorLiteralInstant() {
+		final Instant value = Instant.ofEpochSecond(1234567890, 12);
+		Literal literal = literal(value);
+
+		assertThat(literal).isNotNull();
+		assertThat(literal.temporalAccessorValue()).isEqualTo(value);
+		assertThat(literal.getLabel()).isEqualTo(value.toString());
+		assertThat(literal.getDatatype()).isEqualTo(XSD.DATETIMESTAMP);
+		assertThat(literal.getCoreDatatype()).isEqualTo(CoreDatatype.XSD.DATETIMESTAMP);
```

Comment:
What's the reasoning for using dateTimeStamp instead of just dateTime? As far as I'm aware the only difference is that the time zone is not optional in dateTimeStamp, but dateTime seems to be more far more commonly used.

Reply:
Unfortunately a bit of a hack... currently the DateTimeFormatter is selected based on the XSD type.

So for xsd:dateTime a DateTimeFormatter is selected that uses ChronoField.YEAR field etc, which is not supported by Instant...

Giving it a second thought.... Perhaps it could be implemented the other way round (always use a formatter using Chronofield.INSTANT_SECONDS for xsd:dateTime), though that would break if a Temporal would not support that field

Reply:
@barthanssens but regardless of which formatter you use, you could use the `xsd:dateTime` datatype URI for the resulting literal I think? Every `dateTimeStamp` is also a valid `dateTime`...

Reply:
I have to re-check the code, but I was under the impression that the current code first selects the xsd type on the chronofields of the java class, (where Instant does not support e.g. Year) 
and then that xsd type is used to select the dateformatter

Instead of first do the formatting and then set the xsd datatype.

I'll look into it, I do agree that using dateTime for all cases is a better solution...